doc/adr/0002-use-neo4j-as-persistence.md

```diff
@@ -0,0 +1,19 @@
+# 2. Use Neo4j as Persistence
+
+Date: 2020-06-06
+
+## Status
+
+Accepted
+
+## Context
+
+We need to save information into a persistence layer in order to query and explore the data. The data we will be working with heavily relies on describing relationships which is well suited for a graph like structure.
+
+## Decision
+
+We will use Neo4J as the persistence layer. It is a fairly popular graph database, well documented and mature product.
+
+## Consequences
```

Comment:
Probably worth noting that as a consequence, deploying topo is going to be more complex.  And neo4j might not be in the 'approved technologies' list for some orgs who might consider using topo in future, but that's a future problem. 

I think of topo as a 'reference implementation' - a demo, not necessarily for an organisation to just use as-is.  So the benefit of having a graph db is enough to be worth the extra complexity.

Reply:
Yeah I see. 
A future idea maybe to see if we could create different resolvers for different data stores?
https://www.apollographql.com/docs/apollo-server/data/data-sources/#community-data-sources

There could be like a `topo-sql-adapter` that could plug in, maybe configurable instead?.. The Neo4J could be the default perhaps.

Reply:
Yeah possibly.  We'll probably weld ourselves pretty firmly to cypher and graph queries, but yeah could be reimplemented.

Another useful thing about neo4j is being able to just do arbitrary graph queries directly against the database using the neo4j browser, including the visual graph explorer widgety thing.  Will make exploring use cases faster.

Reply:
Yep, gotcha. It would seem like a decent effort to port cypher to sql but might be worth it to make it more adoptable. (at the expense of exploration utility)